libraries/Microsoft.Bot.Connector/Authentication/AdalAuthenticator.cs

```diff
@@ -217,6 +221,9 @@ private bool IsAdalServiceUnavailable(Exception ex)
             return adalServiceException.ErrorCode == MsalTemporarilyUnavailable || adalServiceException.StatusCode == 429;
         }
 
+        private bool IsAdalServiceInvalidRequest(Exception ex)
+            => ex is AdalServiceException adal && adal.StatusCode == 400;
```

Comment:
Can we use the constants in HttpStatus codes here? I don't think we need any new dependencies for it. Since you are at it, would love to replace the hardcoded `429` online 221 :)

Reply:
There isn't a HttpStatusCode for 429 defined, so I placed it in a constant.